### ISSUE TITLE
keras.ops.cross returns incorrect symbolic output shape

### ISSUE THREAD

**Issue by AGFACBNNR** [keras-team/keras#22525]
`keras.ops.cross()` produces a different output shape for Keras symbolic tensors than for eager TensorFlow tensors when `axis=None`.

### Reproduction steps

```python
import keras
import tensorflow as tf


def call_func(inputs, axis=None):
    x1, x2 = inputs
    return keras.ops.cross(x1, x2, axis=axis)

x1 = tf.ones((3, 4, 3))

x2 = tf.ones((3, 4, 3))

print(call_func([x1, x2], axis=None).shape)
print(call_func([keras.Input(shape=(4, 3)), keras.Input(shape=(4, 3))], axis=None).shape)
```

### Expected result

Both eager and symbolic inputs should produce the same output shape, which is `(3, 4, 3)` for this case.

### Actual result

- Eager tensor output shape: `(3, 4, 3)`
- Keras Input output shape: `(None, 3, 4)`

**Comment by divyashreepathihalli** [comment]
@keras-team /triage

**Comment by divyashreepathihalli** [comment]
@keras-team /triage
